codec/codec.go

```diff
@@ -113,25 +118,32 @@ func (s *ServerCodec) ReadRequestBody(obj interface{}) error {
 // the response was invalid, the size of the body of the resp is reported as
 // having size zero and is not sent.
 func (s *ServerCodec) WriteResponse(resp *rpc.Response, obj interface{}) error {
-	pb, ok := obj.(proto.Message)
-	if !ok {
-		return fmt.Errorf("%T does not implement proto.Message", obj)
-	}
+	// If we can't actually do the cast it means that (most likely) obj is nil,
+	// indicating that we don't actually have a response to send back (e.g. it's
+	// a "void" method call). This is OK because the metadata about the response
+	// object will indicate that everything is A-OK.
+	pb, shouldWriteResponse := obj.(proto.Message)
 
 	// Write the header
 	header := wire.Header{
 		Method: &resp.ServiceMethod,
 		Seq:    &resp.Seq,
 	}
+
 	if resp.Error != "" {
 		header.Error = &resp.Error
 	}
+
 	if err := WriteProto(s.w, &header); err != nil {
 		return nil
 	}
 
 	// Write the proto
-	return WriteProto(s.w, pb)
+	if shouldWriteResponse {
```

Comment:
It's nitpicky obviously, but I tend to put less-common cases inside the if statement;

```Go
if !shouldWriteResponse {
  return nil
}
return WriteProto(s.w, pb)
```